app/models/system-state.server.ts

```diff
@@ -29,11 +29,12 @@ export function getIsReconciliationNeeded(): Promise<SystemState['reconciliation
     .then((data) => data?.reconciliationNeeded ?? true);
 }
 
-export function setIsReconciliationNeeded(
+export async function setIsReconciliationNeeded(
   reconciliationNeeded: SystemState['reconciliationNeeded']
 ) {
+  let result;
   try {
-    return prisma.systemState.update({
+    result = await prisma.systemState.update({
```

Comment:
Could you explain why not returning directly?

Reply:
Before, try block wasn't catching a serialization error when the `update` fails to find the row. When I checked test coverage, the line (47 or 48) that calls `initialize()` was not covered.

I think what happened before was that `update` itself returns the error as shown in [documentation](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#return-type-4), and the return statement returns it , making it land outside of the try block, so catch block was not properly triggered to run `initialize()` .
With this change, I can get catch block to trigger and run `initialize()`.

Reply:
Wouldn't this change affect the places where this function is being called?

Reply:
I think we can simplify this in a way similar to how `getIsReconciliationNeeded` is setup:
```ts
  return prisma.systemState
    .update({
      data: { reconciliationNeeded },
      where: { unique: StateEnumType.unique },
    })
    .catch(() => {
      return initialize();
    });
```

Reply:
See also #493